CredentialProvider.Microsoft/CredentialProviders/Vsts/MSAL/MsalTokenProvider.cs

```diff
@@ -166,32 +229,88 @@ public async Task<IMsalToken> AcquireTokenWithWindowsIntegratedAuth(Cancellation
 
                 throw;
             }
-            finally
-            {
-                var helper = await GetMsalCacheHelperAsync();
-                helper?.UnregisterCache(publicClient.UserTokenCache);
-            }
-       }
+        }
 
-        private async Task<IPublicClientApplication> GetPCAAsync(bool useLocalHost = false)
+        private async Task<IPublicClientApplication> GetPCAAsync(bool useLocalHost)
         {
             var helper = await GetMsalCacheHelperAsync().ConfigureAwait(false);
 
             var publicClientBuilder = PublicClientApplicationBuilder.Create(this.clientId)
-                .WithAuthority(this.authority);
-
-            if (useLocalHost)
-            {
-                publicClientBuilder.WithRedirectUri("http://localhost");
-            }
-            else
+                .WithAuthority(this.authority)
+                .WithLogging(
+                    (LogLevel level, string message, bool _containsPii) =>
+                    {
+                        // We ignore containsPii param because we are passing in enablePiiLogging below.
+                        this.Logger.Verbose($"MSAL Log ({level}): {message}");
+                    },
+                    enablePiiLogging: EnvUtil.GetLogPIIEnabled()
+                );
+            
+            if (this.brokerEnabled)
             {
-                publicClientBuilder.WithRedirectUri(NativeClientRedirect);
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    this.Logger.Verbose($"MSAL using WithBrokerPreview");
+                    publicClientBuilder = publicClientBuilder.WithBrokerPreview();
+
+                    //  needed for WAM
+                    publicClientBuilder = publicClientBuilder.WithParentActivityOrWindow(() => GetConsoleOrTerminalWindow());
+
+                    publicClientBuilder = publicClientBuilder.WithWindowsBrokerOptions(new WindowsBrokerOptions() {
+                        HeaderText = "Azure DevOps Artifacts",
+                        // System.NotImplementedException: The new broker implementation does not yet support Windows account discovery (ListWindowsWorkAndSchoolAccounts option)
+                        // https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3602
+                        // ListWindowsWorkAndSchoolAccounts = true,
```

Comment:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3602 has been merged last month. The first release is [4.49.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.49.0), 4.49.1 is currently the latest.